### ISSUE TITLE
USWDS-Site - Bug: Mobile navigation inaccessible to screen reader touch

### ISSUE THREAD

**Issue by amyleadem** [uswds/uswds-site#1728]
### Describe the bug

The content inside the mobile header is inaccessible to users navigating the site by screen reader touch (VoiceOver Gestures or TalkBack). This happens because `aria-hidden` is applied to the `.usa-nav` element. 

## Details

From the original issue report: https://github.com/uswds/uswds/issues/4787

> In https://github.com/uswds/uswds/issues/3527, an issue was raised where the mobile navigation dialog was correctly trapping focus, but when using VoiceOver the focus could escape the mobile navigation dialog. This was fixed by adding an aria-hidden attribute to everything except the .usa-header element (but only if that header element is a direct descendant of <body>).

> However, this breaks in 2 scenarios, both of which are being demonstrated on https://designsystem.digital.gov/.

> On https://designsystem.digital.gov/ itself, the mobile navigation contents, .usa-nav, is not a child of the .usa-header element, so .usa-nav is incorrectly getting an aria-hidden attribute. The result is that the focus correctly moves through the mobile navigation dialog when using the TAB key, but it does not when using VoiceOver to view the items in the navigation (e.g. VoiceOver Web Item rotor).

### Steps to reproduce the bug

From the original issue report: https://github.com/uswds/uswds/issues/4787
1. Navigate to https://designsystem.digital.gov/
1. Shrink the page, or zoom in to the point where the mobile navigation styling is activated
1. Tab to and activate the menu button
1. In the developer tools element inspector, note that the <header class="usa-header"> does not have an aria-hidden attribute, but note that the <nav class="usa-nav site-nav sidenav-mobile"> does have an aria-hidden attribute, despite being the content that the user is trying to interact with.

### Expected Behavior

All of the page content has aria-hidden applied to it, EXCEPT for the mobile navigation dialog's content.

This should be the case for a variety of .usa-header and .usa-nav document structures, since there are likely many different structures in the wild at this point.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.
